internal/dockerauth/api.go

```diff
@@ -64,32 +69,31 @@ func ParseScope(s string) ([]ResourceAccessRights, error) {
 		ras = append(ras, ra)
 	}
 	if len(errs) > 0 {
-		return ras, &ScopeParseError{errs}
+		return ras, &scopeParseError{errs}
 	}
 	return ras, nil
 }
 
-// A ScopeParseError is an error returned when part of a scope is not
+// A scopeParseError is an error returned when part of a scope is not
 // valid.
-type ScopeParseError struct {
-	Errs []error
+type scopeParseError struct {
```

Comment:
Now that this is unexported, there's even less use for it AFAICS. You could just return the first error.

Reply:
I'lll remove this in a follow up